packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3212 @@
+  /// Private constructor because menu entries can only be created by
+  /// [createMaterialMenu].
+  MenuHandle._({
+    _MenuHandleBase? parent,
+    required List<Widget> menuChildren,
+    FocusNode? buttonFocusNode,
+    VoidCallback? onOpen,
+    VoidCallback? onClose,
+    Offset alignmentOffset = Offset.zero,
+    Offset? globalMenuPosition,
+    ButtonStyle? buttonStyle,
+    MenuStyle? menuStyle,
+    Clip menuClipBehavior = Clip.none,
+    bool ownsParent = false,
+  })  : _parent = parent,
+        _menuChildren = menuChildren,
+        _buttonFocusNode = buttonFocusNode,
+        _onOpen = onOpen,
+        _onClose = onClose,
+        _alignmentOffset = alignmentOffset,
+        _globalMenuPosition = globalMenuPosition,
+        _buttonStyle = buttonStyle,
+        _menuStyle = menuStyle,
+        _menuClipBehavior = menuClipBehavior,
+        _ownsParent = ownsParent,
+        _menuScopeNode = FocusScopeNode() {
+    assert(() {
+      _menuScopeNode.debugLabel = 'Menu Scope';
+      return true;
+    }());
+    _parent?._addChild(this);
+  }
+
+  @override
+  final _MenuHandleBase? _parent;
+
+  /// Returns true if this [MenuHandle]'s menu is open.
+  @override
+  bool get isOpen => _overlayEntry != null;
+
+  @override
+  Axis get _orientation => Axis.vertical;
+
+  final List<Widget> _menuChildren;
+  final FocusScopeNode _menuScopeNode;
+  final MenuStyle? _menuStyle;
+  final ButtonStyle? _buttonStyle;
+  final VoidCallback? _onOpen;
+  final VoidCallback? _onClose;
+  final FocusNode? _buttonFocusNode;
+  final Clip _menuClipBehavior;
+  final Offset _alignmentOffset;
+  final bool _ownsParent;
+  Offset? _globalMenuPosition;
+
+  @protected
+  OverlayEntry? _overlayEntry;
+
+  FocusNode? get _firstItemFocusNode {
+    assert(ChangeNotifier.debugAssertNotDisposed(this));
+    if (_menuScopeNode.context == null) {
+      return null;
+    }
+    final FocusTraversalPolicy policy =
+        FocusTraversalGroup.maybeOf(_menuScopeNode.context!) ?? ReadingOrderTraversalPolicy();
+    return policy.findFirstFocus(_menuScopeNode, ignoreCurrentFocus: true);
+  }
+
+  /// Open the menu, optionally at a global position.
+  ///
+  /// Call this when the menu should be shown to the user.
+  ///
+  /// The optional `position` argument will update the global coordinate where
+  /// the menu should be opened, taking into account the `alignmentOffset`, and
+  /// overriding any `globalMenuPosition` given to [createMaterialMenu].
+  ///
+  /// If `position` is not given, and no `globalMenuPosition` was given to
+  /// [createMaterialMenu], the menu will appear at the location that is
+  /// `alignmentOffset` away from the alignment origin specified by the
+  /// combination of the [MenuStyle.alignment] for the menu and the [MenuAnchor]
+  /// for the menu.
+  void open(BuildContext context, {Offset? position}) {
+    assert(ChangeNotifier.debugAssertNotDisposed(this));
+    if (isOpen && position == _globalMenuPosition) {
```

Comment:
what if it is given a different context - which could result in a different position?

Reply:
Maybe it would be easier to define that an open menu cannot be opend again?